server/routes/teacherRegistration.js

```diff
@@ -0,0 +1,58 @@
+import express, { json } from 'express';
+import User from "../db/schemas/user.js";
+const teacherRouter = express.Router();
+teacherRouter.use(express.json());
+teacherRouter.use(express.urlencoded({extended: true})); 
+
+
+async function validateId(user_id) {
+    try {
+      const user = await User.findOne({ user_id: user_id });
+      return user ? false : true; 
+    } catch (error) {
+      console.error(error);
+      return false; 
+    }
+  }
+
+function validatePassword(pw){
+    return /[A-Z]/       .test(pw) &&
+           /[a-z]/       .test(pw) &&
+           /[0-9]/       .test(pw) &&
+           /[^A-Za-z0-9]/.test(pw) &&
+           pw.length > 11;
+}
+
+teacherRouter.post("/", async(req,res) => {
```

Comment:
this is not a blocker and should be in another PR, but the route should be defined here and not in api.js